### PR TITLE
Layered: Fix wrong hierarchy handling. #700

### PR DESCRIPTION
**Pull request by le-cds**
Signed-off-by: le-cds <zephanya@web.de>

### CODE REVIEW

File: plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/graph/transform/ElkGraphImporter.java [hunk 1]
```diff
@@ -342,73 +341,75 @@ private void importHierarchicalGraph(final ElkNode elkgraph, final LGraph lgraph
...
+                if (parentLNode != null) {
+                    parentLGraph = parentLNode.getNestedGraph();
+                }
+                
+                // Transform the edge, finally...
+                LEdge ledge = transformEdge(elkedge, parentElkGraph, parentLGraph);
+                
+                // Find the graph the edge's coordinates will have to be made relative to during export
+                LGraph coordinateSystemOrigin = findCoordinateSystemOrigin(elkedge, elkgraph, lgraph);
+                if (coordinateSystemOrigin != null) {
```
Comment (uruuru) [inline plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/graph/transform/ElkGraphImporter.java:396]: If it's `null`, it's assumed to be already in the correct coordinate system?

**Comment by uruuru** [comment]
The new code makes sense to me. However, since it's hard to diff it, could you briefly state what the issue supposedly was?

File: plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/graph/transform/ElkGraphImporter.java (hunk 1 above)
Comment (le-cds) [reply plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/graph/transform/ElkGraphImporter.java:396]: Yep, but that's an unnecessary distinction, I guess. If it's `null`, the `setProperty(...)` call wouldn't change anything anyway, so I'll just remove the conditional statement there.

**Comment by le-cds** [comment]
Sure. In broad terms, the problem goes back to [this remark of yours](https://github.com/eclipse/elk/commit/66f943f995f521795e78b23e01bf9de5ce0a2396#r40762087): we should iterate over _contained_, not _outgoing_ edges. That makes it a lot easier to keep track of which edges actually need to be transformed, and which don't.

**Pull request merged**